MaxText/tests/forward_pass_logit_checker.py

```diff
@@ -111,9 +111,22 @@ def main(config, test_args):
     max_logging.log(f"{full_train_logits[0, 0, :]=}")
     token_size = int(test_args.token_size) if test_args.token_size else golden_logits.shape[0]
     max_logging.log(f"Max Difference {np.max(np.subtract(full_train_logits[0, :token_size, :], golden_logits[:token_size, :]))}")
-    assert jax.numpy.allclose(
+
+    model_probabilities = jax.nn.softmax(full_train_logits[0, :token_size, :], axis=-1)
+    golden_probabilities = jax.nn.softmax(golden_logits[:token_size, :], axis=-1)
+
+    max_logging.log(f"{golden_probabilities[0]=}")
+    max_logging.log(f"{model_probabilities[0]=}")
+
+    kl_div = jax.numpy.sum(jax.scipy.special.kl_div(golden_probabilities, model_probabilities), axis=-1)
+    max_logging.log(f"{kl_div=}")
+
+    if test_args.kl is not None:
+      assert jax.numpy.all(kl_div < test_args.kl)
```

Comment:
Should we name it `max_kl_divergence` or `max_kl_div` and have a comment like:

f"KL Divergence ({kl_div}) exceeds the allowed threshold "
            f"({test_args.max_kl_divergence})"

Reply:
+1, clear error message very appreciated

Reply:
Thanks for the suggestions, updated the logging 
http://shortn/_XlXElFUgnv added the logs of 4 sample tests here (kl passed, kl failed, numerical passed, numerical failed) with updated logging message